tests/e2e/tests/pilot/pilot_test.go

```diff
@@ -220,16 +220,44 @@ func runRetriableTest(t *testing.T, cluster, testName string, retries int, f fun
 	})
 }
 
+type resource struct {
+	// Kind of the resource
+	Kind string
+	// Name of the resource
+	Name string
+}
+
 // deployableConfig is a collection of configs that are applied/deleted as a single unit.
 type deployableConfig struct {
-	Namespace  string
-	YamlFiles  []string
+	Namespace string
+	YamlFiles []string
+	// List of resources must be removed during deployableConfig setup, and restored
+	// during teardown. These resources must exist before deployableConfig setup runs, and should be
+	// in the same namespace defined above. Typically, they are added by the default Istio installation
+	// (e.g the default global authentication policy) and need to be modified for tests.
+	Removes    []resource
```

Comment:
So the `Removes` must be removed in the beginning and added back in the end? I understand this applies to global default permissive policy, but still feel a bit of uncommon as general test setup. Can we just inline these in auth test itself instead of making it generic?

Reply:
This is cherry-picked from master, so I don't want to make a change. Sure the framework can be improved, but let's do that in master later,